library/kani_macros/src/lib.rs

```diff
@@ -184,6 +234,110 @@ mod sysroot {
         }
     }
 
+    /// Rewrites function contract annotations (`requires`, `ensures`) by lifing
+    /// the condition into a separate function. As an example: (using `ensures`)
+    ///
+    /// ```rs
+    /// #[kani::ensures(x < result)]
+    /// fn foo(x: u32) -> u32 { .. }
+    /// ```
+    ///
+    /// Is rewritten to
+    ///
+    /// ```rs
+    /// fn foo_enusres_<hash of foo>(x: u32, result: u32) {
```

Comment:
I checked this indirectly in the `vec-map` example, which uses multiple `ensures` on a single function and it does not complain about duplicate functions so I can only assume that the hash changes. (would make sense too, since the `ItemFn` contains the attributes of the function)

Reply:
Can you please add a second ensures to one of your tests?

Reply:
Done